dist/angular-selector.js

```diff
@@ -375,9 +383,10 @@
 					if (!scope.multiple) scope.selectedValues = (scope.options || []).filter(function (option) { return scope.optionEquals(option); }).slice(0, 1);
 					else
 						scope.selectedValues = (scope.value || []).map(function (value) {
-							return $filter('filter')(scope.options, function (option) {
+							var inList = $filter('filter')(scope.options, function (option) {
 								return scope.optionEquals(option, value);
 							})[0];
+							return scope.createCustom ? value : inList;
```

Comment:
You don't need this, you just have to use `scope.set(option)` function (passing your new option as an argument). Check out my last comment [here](https://github.com/indrimuska/angular-selector/pull/11#issuecomment-176249140):
```javascript
            scope.options.push(option);
            scope.set(option); // <--------- use this
```